.github/workflows/rust.yml

```diff
@@ -157,8 +160,26 @@ jobs:
           cairo_programs/**/*.json
         key: cairo_1_test_contracts-cache-${{ hashFiles( 'cairo_programs/**/*.cairo' ) }}
 
-    - name: Check all features
-      run: cargo check-all-features --workspace --all-targets
+    # NOTE: we do this separately because --workspace operates in weird ways
+    - name: Check all features (felt)
+      run: |
+        cd felt
+        cargo check-all-features
+        cargo check-all-features --workspace --all-targets
+
+    - name: Check all features (vm)
+      run: |
+        cd vm
+        cargo check-all-features
+
+    - name: Check all features (CLI)
+      run: |
+        cd cairo-vm-cli
+        cargo check-all-features
+    
+    - name: Check all features (workspace)
+      run: |
+        cargo check-all-features --workspace --all-targets
```

Comment:
Maybe a strategy matrix would work better here? I think you can use the `-p` flag.

Reply:
I thought about that, but using a matrix wouldn't work here as the four steps share a lot of compiled artifacts. And for the `-p` flag, it works the same way as `--workspace` or no flag (it compiles the whole workspace).